cpp_exercise/src/lib/filehandlerlib.cc

```diff
@@ -0,0 +1,176 @@
+#include "filehandlerlib.h"
+#include "ipcexceptionlib.h"
+
+/**
+ *
+ * @brief FileHandler Constructor for saving the file name
+ * 
+ * 
+ */
+
+FileHandler::FileHandler(const std::string & filename) : _file_name(filename){
+
+}
+
+/**
+ * @brief Destroy the File Handler:: File Handler object and close the file 
+ * 
+ */
+
+FileHandler::~FileHandler(){
+
+    close(_fd);
+    
+}
+
+
+/**
+ * @brief Creating a file
+ * 
+ */
+void FileHandler::createFile(){
+
+    
+   
+    _fd = open(_file_name.c_str(), O_RDWR| O_CREAT | O_TRUNC, _mode);
+    if(_fd<0){
+       std::cout<<strerror(errno)<<std::endl;
+        throw IPCException("FileHandler ERROR: Cannot Open the File Correctly");
+    }
+
+
+}
+
+/**
+ * @brief Writing  a vector<unsigned char> into a file
+ * 
+ */
+
+void FileHandler::writeFile(const std::vector<unsigned char> & file_input, size_t write_size){
+
+    
+     _write_size = write_size;
```

Comment:
I don't understand you copy this size_t. If I am not wrong, you use it only in this method, so you can take the parameter directly.

Reply:
I thought that maybe it is a good practice to save the parameter into its private variables. But I am not sure about this

Reply:
It is if you need it after. But in this case you won't.

Reply:
Fixed